### PR TITLE
Add MUSINFO support

### LINKED ISSUE #123: suggestion: support for MUSINFO
I want to integrate MUSINFO support from Crispy. It seems the only non-trivial part is the extension to savegame.

### PR DESCRIPTION
**Pull request by rfomin**
Fixes #123

### CODE REVIEW

File: Source/g_game.c [hunk 1]
```diff
@@ -1518,6 +1520,11 @@ static void G_DoSaveGame(void)
   memcpy(save_p, &totalleveltimes, sizeof totalleveltimes);
   save_p += sizeof totalleveltimes;
 
+  // save lump name for current MUSINFO item
+  CheckSaveGame(9);
+  strncpy(save_p, lumpinfo[musinfo.current_item].name, 9);
```
Comment (fabiangreffrath) [inline Source/g_game.c:1525]: better use `memcpy()` here

**Review by fabiangreffrath — APPROVED** [review]

File: Source/g_game.c [hunk 2]
```diff
@@ -1634,6 +1641,25 @@ static void G_DoLoadGame(void)
     save_p += sizeof totalleveltimes;
   }
 
+  // restore MUSINFO music
+  if (save_p - savebuffer <= length - 8)
+  {
+    char lump[9];
+    int i;
+
+    memcpy(lump, save_p, 8);
```
Comment (fabiangreffrath) [inline Source/g_game.c:1650]: Now we need to make sure that `lump[8]` is `\0`.

**Review by fabiangreffrath — APPROVED** [review]

File: Source/g_game.c (hunk 2 above)
Comment (fabiangreffrath) [reply Source/g_game.c:1650]: You could e.g. initialize `char lump[9] = {0};`.

Comment (rfomin) [reply Source/g_game.c:1650]: Yes, I know. I was distracted.

Comment (fabiangreffrath) [reply Source/g_game.c:1650]: No problem, looks good now!

File: Source/g_game.c [hunk 3]
```diff
@@ -35,6 +35,7 @@
 #include "f_finale.h"
 #include "m_argv.h"
 #include "m_misc.h"
+#include "m_misc2.h"
```
Comment (fabiangreffrath) [inline Source/g_game.c:38]: Now I guess we don't need this anymore? Sorry, small steps...

**Comment by fabiangreffrath** [comment]
Thank you very much!

**Pull request merged**